File: SpringBoot/springboot-java-h2-sample/src/main/java/com/se/example/springbootjavah2/service/CustomerServices.java
package com.se.example.springbootjavah2.service;


import com.se.example.springbootjavah2.entities.Customer;
import com.se.example.springbootjavah2.repository.CustomerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.List;

@Service
public class CustomerServices {
    @Autowired
    CustomerRepository customerRepository;

    public void deleteAll() {
        customerRepository.deleteAll();
    }

    public void save(Customer customer) {
        customerRepository.save(customer);
        customerRepository.flush();
    }

    @Transactional
    public void showAll() {
        List<Customer> custs = customerRepository.findAll();
        custs.forEach(System.out::println);
    }
}

File: old/SpringBoot/springboot-kotlin-h2-sample/src/main/kotlin/com/se/example/springbootkotlinh2sample/service/CustomerServices.kt
package com.se.example.springbootkotlinh2sample.service

import org.springframework.stereotype.Service
import com.se.example.springbootkotlinh2sample.repositories.CustomerRepository
import org.springframework.beans.factory.annotation.Autowired
import com.se.example.springbootkotlinh2sample.entities.Customer
import java.util.function.Consumer
import javax.transaction.Transactional


@Service
class CustomerServices {

    @Autowired
    lateinit var customerRepository: CustomerRepository

    fun deleteAll() {
        customerRepository.deleteAll()
    }

    fun save(customer: Customer) {
        customerRepository.save(customer)
        customerRepository.flush()
    }

    @Transactional
    fun showAll() {
        val custs = customerRepository.findAll()
        custs.forEach(Consumer<Customer> { println(it) })
    }


}